example_project_app/plugins.py

```diff
@@ -211,6 +226,60 @@ class ProjectAppPlugin(ProjectModifyPluginMixin, ProjectAppPluginPoint):
             'default': '',
             'description': 'Example json project user setting',
         },
+        'project_callable_setting': {
+            'scope': SODAR_CONSTANTS['APP_SETTING_SCOPE_PROJECT'],
+            'type': 'STRING',
+            'label': 'Callable project setting',
+            'default': get_example_setting_default,
+            'description': 'Example callable project setting',
+        },
+        'user_callable_setting': {
+            'scope': SODAR_CONSTANTS['APP_SETTING_SCOPE_USER'],
+            'type': 'STRING',
+            'label': 'Callable user setting',
+            'default': get_example_setting_default,
+            'description': 'Example callable user setting',
+        },
+        'project_user_callable_setting': {
+            'scope': SODAR_CONSTANTS['APP_SETTING_SCOPE_PROJECT_USER'],
+            'type': 'STRING',
+            'default': get_example_setting_default,
+            'description': 'Example callable project user setting',
+        },
+        'project_callable_setting_options': {
+            'scope': SODAR_CONSTANTS['APP_SETTING_SCOPE_PROJECT'],
+            'type': 'STRING',
+            'label': 'Callable setting with options',
+            'default': get_example_setting_default,
+            'options': [
```

Comment:
Here we should have just one method returning the list of options. Also, we should set up a separate example method to provide the example options. Note what I wrote in the spec, the return value must be a list of either direct values or tuples of `(value, label)`. Support for both must be implemented and tested, of course.

Please update this and related usage & tests.